### PR TITLE
Ew20 btg

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+---
+title: Boot Guard - pre-execution firmware verification
+abstract: 'This post will not describe how to guard your shoes. However,
+          will definitely introduce you to Boot Guard feature present on
+          Intel processors which allows firmware verification before the first
+          instruction executes. One may call it pre-execution firmware
+          verification.'
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 Staszek will use that description for social media so maybe we should mention coreboot, open source firmware and other keywords?

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+application or operating system. So Boot Guard means that it protects the boot
+process and the firmware. But from what should it protect and why? You can find
+it out by spending few spare minutes and reading this post.
+
+![Boot + Guard](/covers/boot_guard.png)
+
+---
+### Why Boot Guard?
+
+Nowadays security researchers are very focused on trustful solutions, security
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]:
```suggestion
Nowadays security researchers are very focused on trustworthy solutions, security
```

Trusted?

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+technologies and low-level safety of the computer machines used everyday. They
+explore every possibility and attack vector to find vulnerabilities in the most
+crucial parts of security in modern systems, like: Intel ME, UEFI etc. By
+exposing discovered threats the silicon vendors release fixes to protect the
+hardware and software from being compromised.
+
+How do you know that software you use is trustful, secure and hasn't been
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]:
```suggestion
How do you know that software you use is trustworthy, secure and hasn't been
```

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+maliciously modified before execution? One can only ensure that by establishing
+the Root of Trust (ROT) as early as possible. There are 2 main type of ROTs:
+
+- SRTM - Static Root of Trust for Measurement
+- DRTM - Dynamic Root of Trust for Measurement
+
+I will not explain the measurement process as it is not a part of this post.
+What I will focus on is cryptographical verification in the ROT. ROT is the
+beginning of the verification process, it is considered to be trusted and
+verifies all components executed by firmware.
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 how this connects with mentioned acronyms? Can you point to DRTM explanation or other TrenchBoot blog/conference paper?

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 1 above)
Comment (pietrushnic) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 also in title we should mention Protectli as well as in description

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+One has to either pay thousands of $ to some big firmware development companies
+to enable Boot Guard on your hardware or buy a high end machine which has Boot
+Guard enabled.
+
+That was true up till recently, where one of our customers -
+[Protectli](https://protectli.com/) - asked us to improve the security of his
+products called Vault with Boot Guard. Protectli sells compact network
+appliance devices running Intel processors. The most powerful of them have
+Intel 7th generation processors which are Boot Guard capable. As a matter of
+fact that 3mdeb did coreboot port for those platforms to coreboot, it was even
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 I believe logic is little bit messed in this sentence

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+easier to experiment with Boot Guard. And finally we have succeed. First
+official platform running open-source firmware that has Boot Guard enabled is
+here. It was possible thanks to the fact that Protectli cares about freedom and
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 "First official platform running open-source firmware that has Boot Guard enabled is here." this is great marketing sentence.

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+see value in open-source firmware and software.
+
+One will be able to witness these results at our booth at Embedded World
+2020, but now I will give a small foretaste what awaits those interested at
+Embedded World 2020 exhibition.
```
Comment (pietrushnic) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 we need Protectli logo and EW2020 banner here to show where we are

File: blog/content/post/2020-02-18-ew2020-btg-demo.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+---
+### Where can I get Boot Guard for my platform?
+
+One has to either pay thousands of $ to some big firmware development companies
+to enable Boot Guard on your hardware or buy a high end machine which has Boot
+Guard enabled.
+
+That was true up till recently, where one of our customers -
+[Protectli](https://protectli.com/) - asked us to improve the security of his
```
Comment (andyhhp) [inline blog/content/post/2020-02-18-ew2020-btg-demo.md]: "one of our customers" => "the security of their products" (not his.)

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 1 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Done

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 4 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Added links few line below. Something like that will be ok?

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 5 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: too much coreboot in coreboot, fixed

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 6 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Indeed

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 7 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Added

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 8 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Yes, I have noticed it when I read it again. Thank you. Fixed.

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 3 above)
Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Suggestion commited

File: blog/content/post/2020-02-18-ew2020-btg-demo.md (hunk 4 above)
Comment (pietrushnic) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: @miczyg1 ok, but you should probably directly say that Intel Boot Guard is SRTM implementation.

Comment (miczyg1) [reply blog/content/post/2020-02-18-ew2020-btg-demo.md]: Done

**Pull request merged**